src/detect-reference.c

```diff
@@ -132,12 +132,15 @@ static DetectReference *DetectReferenceParse(const char *rawstr, DetectEngineCtx
         goto error;
     }
 
-    if (strlen(key) == 0 || strlen(content) == 0)
+    int ref_len = strlen(content);
+    if (strlen(key) == 0 || ref_len == 0)
         goto error;
 
     SCRConfReference *lookup_ref_conf = SCRConfGetReference(key, de_ctx);
     if (lookup_ref_conf != NULL) {
         ref->key = lookup_ref_conf->url;
+        /* already bound checked to be REFERENCE_CONTENT_NAME_MAX or less */
```

Comment:
nit-ish: the comments about the bound checks are mismatched, here, right? (i.e., here it should be `REFERENCE_CONTENT_SYSTEM_MAX` )

Reply:
heh, just noticed that I typo'ed when adding my review. sheesh >__<'